presto-hive/src/main/java/com/facebook/presto/hive/HiveMaterializedViewUtils.java

```diff
@@ -78,7 +79,7 @@ public static void validateMaterializedViewPartitionColumns(
     {
         SchemaTableName viewName = new SchemaTableName(viewTable.getDatabaseName(), viewTable.getTableName());
 
-        Map<String, Map<SchemaTableName, String>> viewToBaseColumnMap = viewDefinition.getColumnMappingsAsMap();
+        Map<String, Map<SchemaTableName, String>> viewToBaseColumnMap = viewDefinition.getNonNullColumnMappingsAsMap();
```

Comment:
rename this to indicate that it excludes columns of tables on the outer join side?